Simple Incremental/Assets/Scripts/Movement.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class Movement : MonoBehaviour
+{
+
+    public event MovementFinishedHandler OnMovementFinished;
+    public delegate void MovementFinishedHandler();
+
+    public float moveSpeed;
+    
+    [SerializeField] 
+    private Transform target;
+
+    [SerializeField] 
+    private float terminationDistance = 0.01f;
+
+    private bool shouldMove = true;
+    private bool movementFinished = false;
+   
+    public void StartMovement()
+    {
+        movementFinished = false;
+        shouldMove = true;
+    }
+    
+    public void StopMovement()
+    {
+        shouldMove = false;
+    }
+
+    public void Update()
+    {
+        if (!movementFinished && shouldMove)

```

Comment:
Consider using just the sholdMove variable.

If movement finishes, set the shouldMove variable to false. Movement finished is a descriptive variable (explains the game state), but not instructive (gives instructions).

Reply:
Fixed